pkg/tests/testsupport/series_gen.go

```diff
@@ -0,0 +1,82 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package testsupport
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/timescale/promscale/pkg/prompb"
+)
+
+type seriesGen struct {
+	ts []prompb.TimeSeries
+}
+
+const (
+	labelKeyPrefix   = "test_k_"
+	labelValuePrefix = "test_v_"
+	metricNamePrefix = "test_metric_"
+)
+
+var samples = []prompb.Sample{{Timestamp: 1, Value: 1}}
+
+func NewSeriesGenerator(numMetrics, numSeriesPerMetric, labels int) (*seriesGen, error) {
+	if labels < 2 {
+		return nil, fmt.Errorf("minLabels cannot be less than 2")
+	}
+	totalSeries := numMetrics * numSeriesPerMetric
+	ts := make([]prompb.TimeSeries, 0, totalSeries)
+
+	labels -= 1 // Since 1 label will be occupied by __name__
+	labelKeys := make([]string, labels)
+	for i := 0; i < labels; i++ {
+		labelKeys[i] = randomText(labelKeyPrefix)
```

Comment:
Can we use index as a suffix to generate unique values instead of random? Just to avoid the probability of collision which is any way very low :)

Reply:
Other option would be setting up a [constant seed](https://pkg.go.dev/math/rand#Seed), so that there we can bring a predictability. Having a real random value in the test is not a good thing.

Reply:
I believe `math/rand` by default is providing deterministic number generator so we should be good

Reply:
No, by default it generates pseudo random number.

Reply:
```
for i := 0; i < 10; i++ {
		fmt.Printf("%d. %d\n", i, rand.Int())
	}
```
Always generates the same number sequence so it is deterministic.

Reply:
Ah, you are right. Looks like I assumed it is a crypto/rand, but actually it is a math/rand.